pages/docs/support/config-files.mdx

```diff
@@ -0,0 +1,672 @@
+---
+title: "Configuration Files"
+date: "2022-02-09"
+pullQuote: "Definitions and examples of Grouparoo configuration files."
+---
+
+Grouparoo projects contain configuration files for Apps, Models, Sources, Properties, Groups, Destinations, and any other objects is exported into config files.
+These config files can be created and updated with [UI config](/docs/config).
+
+Config files are either JSON or JavaScript which allow you the flexibility to manipulate data or load configuration from external sources.
+
+## Config Files Location
+
+Config files are saved at the root of your Grouparoo project directory, in a directory called `config`. A typical Grouparoo project structure looks like this:
+
+```
+my-grouparoo-project/
+├── package.json
+├── package-lock.json
+├── .env
+└── config/
+    └── ... (config files go here)
+```
+
+You can optionally choose to use another directory for your configuration by setting the `GROUPAROO_CONFIG_DIR` environment variable, for example `GROUPAROO_CONFIG_DIR=/path/to/config`.
+
+## Common Config Attributes
+
+Every type of object has its own definition but all of them require these attributes:
+
+- `id`: (**required**) A unique identifier for this object.
+  - Lowercase letters, numbers, and underscores only
+  - No spaces
+  - Fewer than 38 characters
+  - Must be unique among other objects of the same type
+- `class`: (**required**) The type of object that is being defined such as `App`, `Model`, `Source`, `Property`, `Group`, `Destination`
+- `name`: (**required**) The display name of this object
+
+## App Config
+
+The App configures the connections available to create Sources and Destinations.
+
+- `id`: (**required**) Unique identifier for this app config.
+- `name`: (**required**) A display name. It matches the `id` by default.
```

Comment:
Am I right to think that if the name is assigned in config UI and does not meet the requirements for ID, the name will be altered to assign a valid id? If so, do you think we should we add something like this so users aren't confused about rules for assigning a name in the UI if they read this?

Reply:
I'll remove this because it's related to the generator. In config UI, the name is driven by what the user types and this is more related to how the generator worked